src/jenkins_job_insight/vapid.py

```
@@ -0,0 +1,159 @@
+    Raises ``RuntimeError`` if the file remains empty/corrupt after all
+    retries.
+    """
+    for attempt in range(retries):
+        try:
+            keys = json.loads(key_file.read_text(encoding="utf-8"))
+            if keys.get("public_key") and keys.get("private_key"):
+                return keys
+            raise RuntimeError("missing keys")
+        except (json.JSONDecodeError, RuntimeError, OSError):
+            if attempt == retries - 1:
+                raise RuntimeError(f"VAPID key file {key_file} is empty or corrupt")
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)